cdap-docs/tools/docs-cli-commands.py

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+#
+# Copyright © 2016 Cask Data, Inc.
+#
+# Used to generate a table in the CLI documentation from the output of CLI tools.
+# 
+# If input_file is not found, puts out a warning message in the log and puts a note 
+# in the table that is generated.
+#
+
+from optparse import OptionParser
+import os
+import sys
+
+VERSION = "0.0.1"
+
+LITERAL = '``'
+SPACES = ' ' * 3
+SECTION_LINE = SPACES + '**'
+COMMAND_LINE = SPACES + LITERAL
+LITERAL_LINE = SPACES + ' | '
+
+SKIP_SECTIONS = ['security (beta)']
+
+MISSING_FILE_TEMPLATE = "   **Missing Input File**,\"Missing input file %s\""
+TABLE_HEADER = """.. csv-table::
+   :header: Command,Description
+   :widths: 50, 50
+
+"""
+
+def parse_options():
+    """ Parses args options.
+    """
+    parser = OptionParser(
+        usage="\n\n  %prog input_file output_file",
+        description="Generates an rst table to be included in the documentation")
+
+    parser.add_option(
+        "-v", "--version",
+        action="store_true",
+        dest="version",
+        help="Version of this software",
+        default=False)
+
+    (options, args) = parser.parse_args()
+    
+    if options.version:
+        print "Version: %s" % VERSION
+        sys.exit(1)
+
+    if not args or len(args) != 2:
+        parser.print_help()
+        sys.exit(1)
+
+    return options, args
+
+#
+# Utility functions
+#
+
+def line_starts_with(line, left):
+    return bool(line[:len(left)] == left)
```

Comment:
For some reason, in an earlier project, I was finding that `startswith` was failing, and I don't remember why. I ended up defining this. I can see it it can be dispensed with.